### PR TITLE
POST /tenants

### PR DESCRIPTION
**Pull request by mchalski**
Issues: https://tracker.mender.io/browse/MEN-1544

@mendersoftware/rndity @maciejmrowiec

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -118,6 +118,7 @@ script:
     - if [[ "$JOB_TYPE" = compile_and_acceptance_tests ]]; then
         CGO_ENABLED=0 go test -c -o deviceadm-test -coverpkg $(go list ./... | grep -v vendor | grep -v mocks | grep -v test | tr  '\n' ,);
         sudo docker build -f Dockerfile.acceptance-testing -t mendersoftware/deviceadm:prtest .;
+        go build;
```
Comment (bboozzoo) [inline .travis.yml]: It's already there in line 107

Comment (bboozzoo) [inline .travis.yml]: It's alread there in line 107

File: store/mongo/datastore_mongo_test.go [hunk 2]
```diff
@@ -476,7 +476,7 @@ func TestMigrate(t *testing.T) {
 
 			// set up automigration
 			if tc.automigrate {
-				db = db.WithAutomigrate()
+				db = db.WithAutomigrate().(*DataStoreMongo)
```
Comment (bboozzoo) [inline store/mongo/datastore_mongo_test.go]: Can we somehow skip type assertion here?

File: .travis.yml (hunk 1 above)
Comment (mchalski) [reply .travis.yml]: fixed

File: store/mongo/datastore_mongo_test.go (hunk 2 above)
Comment (mchalski) [reply store/mongo/datastore_mongo_test.go]: no, WithAutomigrate must be public, so part of the interface, so it must return the same interface.

aside from introducing another wrapper maybe, but I didn;t want this to happen.

**Pull request merged**